apps/vanilla/src/pages/black-jack/models/player.ts

```diff
@@ -0,0 +1,52 @@
+import { Subscriber } from '../type/subscriber';
+
+import { Publisher } from './publisher';
+const WIN_SCORE = 21;
+
+/** Player with publisher and subscriber func. */
+export class Player extends Publisher<number> implements Subscriber<number> {
+	/** All dice results for one player. */
+	private readonly diceResults: number[] = [];
+
+	/**
+		* Push result dice roll in general array.
+		* @param dice - Number.
+		*/
+	public result(dice: number): void {
+		this.diceResults.push(dice);
+	}
+
+	/** Get sum of all Dices. */
+	public getScore(): number {
+		return this.diceResults.reduce((acc, dice) => acc + dice, 0);
+	}
+
+	/** Check win status in general array. */
+	public getWinner(): boolean {
+		return this.getScore() >= WIN_SCORE;
+	}
+
+	/** Make format dices for HtmlElements. */
+	public printResult(): string {
+		return this.diceResults.join(' ');
```

Comment:
I think it doesn't matter how these values will be used later. We can improve this JSDoc:
```suggestion
	/** Get result in a user-friendly format. */
	public printResult(): string {
		return this.diceResults.join(' ');
```